contracts/Front/ShelteringTransfers.sol

```diff
@@ -67,20 +81,65 @@ contract ShelteringTransfers is Base {
     }
 
     function getDonor(bytes32 transferId) public view returns(address) {
-        (address donorId, ) = shelteringTransfersStore.getTransfer(transferId);
+        (address donorId,, ) = shelteringTransfersStore.getTransfer(transferId);
         return donorId;
     }
 
     function getTransferredBundle(bytes32 transferId) public view returns(bytes32) {
-        (, bytes32 bundleId) = shelteringTransfersStore.getTransfer(transferId);
+        (, bytes32 bundleId, ) = shelteringTransfersStore.getTransfer(transferId);
         return bundleId;
     }
 
     function transferIsInProgress(bytes32 transferId) public view returns(bool) {
-        (address donorId, ) = shelteringTransfersStore.getTransfer(transferId);
+        (address donorId,, ) = shelteringTransfersStore.getTransfer(transferId);
         return donorId != address(0x0);
     }
 
+    function canResolve(address resolverId, bytes32 transferId) public view returns (bool) {
+        (address donorId, bytes32 bundleId, ) = shelteringTransfersStore.getTransfer(transferId);
+
+        // solium-disable-next-line operator-whitespace
+        return donorId != address(0x0) &&
+        !sheltering.isSheltering(bundleId, resolverId) &&
+        resolverId == getTransferDesignatedShelterer(transferId);
```

Comment:
Why do we want to add DMP into sheltering transfers? The aim of transfers is to get rid of our bundles as soon as possible. Also, I don't understand why introduce `canResolve` when we have `requireResolutionPossible` doing the same thing

Reply:
In response to TransferStarted event, all Atlases will start downloading the bundle from the donor, which will lead to a DDoS attack on it. Therefore, we need to limit the number of possible resolvers, using DMP as the proven and effective solution.

Reply:
requireResolutionPossible method has been removed from ShelteringTransfers because its functionality has been moved to canResolve.

Reply:
Do we have any data that DMP here is necessary? I'd love to see some data backing up this decision.
I mean we can measure how much load a typical (small DO droplet) Atlas server can handle. We can also estimate the number of requests it will get (expected number of Atlases in the network x bundles the donor wants to transfer).

Reply:
Suppose that we have 500 Atlases and an average bundle size of 2 MB (in a cryptoeconomy, the average bundle size is 50 MB) - hence each transfer request will generate 1 GB of traffic. And this is only one transfer request! If requests will be 100? I think that such traffic will completely occupy Atlas transmission channel.

Reply:
Do you agree that DMP is needed here?

Reply:
Yes, I agree.